village/src/App.js

```diff
@@ -1,24 +1,74 @@
+// DEPENDANCIES:
 import React, { Component } from 'react';
+import axios from 'axios';
+import { Route } from 'react-router-dom';
 
-import './App.css';
+// COMPONENTS:
 import SmurfForm from './components/SmurfForm';
 import Smurfs from './components/Smurfs';
+import Home from './components/Home';
+import Navigator from './components/Navigator';
+
+// STYLES:
+import './App.css';
 
+// CLASS COMPONENT:
 class App extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      smurfs: [],
+      smurfs: []
     };
   }
-  // add any needed code to ensure that the smurfs collection exists on state and it has data coming from the server
-  // Notice what your map function is looping over and returning inside of Smurfs.
-  // You'll need to make sure you have the right properties on state and pass them down to props.
+
+  // API GET REQUEST:
+  async componentDidMount() {
+    try {
+      const res = await axios.get('http://localhost:3333/smurfs');
+      this.setState({ smurfs: res.data });
+    } catch (err) {
+      console.log(err.message);
+    }
+  }
+
+  // HANDLERS:
+  handleNewSmurf = newSmurf => {
+    axios
+      .post('http://localhost:3333/smurfs/', newSmurf)
+      .then(res =>
+        this.setState({
+          smurfs: res.data
+        })
+      )
+      .catch(err => console.log(err));
+  };
+
+  deleteSmurf = id => e => {
+    e.preventDefault();
+
+    axios
+      .delete(`http://localhost:3333/smurfs/${id}`)
+      .then(res =>
+        this.setState({
+          smurfs: res.data
+        })
+      )
+      .catch(err => console.log(err));
+  };
+
```

Comment:
handleNewSmurf = newSmurf => {
    axios
      .post('http://localhost:3333/smurfs/', newSmurf)
      .then(res =>
        this.setState({
          smurfs: res.data
        })
      )
      .catch(err => console.log(err));
  };


  deleteSmurf = id => e => {
    e.preventDefault();


    axios
      .delete(`http://localhost:3333/smurfs/${id}`)
      .then(res =>
        this.setState({
          smurfs: res.data
        })
      )
      .catch(err => console.log(err));

well written, readable code.